fe/parameterize_simple.py

```diff
@@ -152,65 +175,148 @@ def close(self):
         for atom_idx, pos in enumerate(guest_conf):
             conformer.SetAtomPosition(atom_idx, (float(pos[0]), float(pos[1]), float(pos[2])))
 
-    # tbd training code
-
-    for host_idx, host_pdb_file in enumerate([args.complex_pdb]):
-
-        host_pdb = app.PDBFile(host_pdb_file)
-
-        # deletion is difficult because of clashes arising from poorly minimized structures
-        # so we need to work on the coupled insertion/deletion schedule
-        # for mode in ['insertion', 'deletion']: 
-        for mode in ['insertion']: 
-
-            print("Mode", mode)
-
-            host_conf = []
-            for x,y,z in host_pdb.positions:
-                host_conf.append([to_md_units(x),to_md_units(y),to_md_units(z)])
-            host_conf = np.array(host_conf)
-
-            # recenter the ligand based on whether we're using the host or the solvent
-            if host_idx == 0:
-                host_name = "complex"
-                conf_center = conf_com
-            elif host_idx == 1:
-                host_name = "solvent"
-                conf_center = com(host_conf)
-
-            init_combined_conf = np.concatenate([host_conf, init_conf])
-
-            # perm = hilbert_sort(init_combined_conf)
-            # perm = np.arange(init_combined_conf.shape[0])
-            sim = simulation.Simulation(
-                guest_mol,
-                host_pdb,
-                mode,
-                step_sizes,
-                cas,
-                lambda_schedule
-            )
-
-            # sample from the rdkit DG distribution (this can be changed later to another distribution later on)
-            all_args = []
-            for conf_idx in range(num_conformers):
-
-                conformer = guest_mol.GetConformer(conf_idx)
-                guest_conf = np.array(conformer.GetPositions(), dtype=np.float64)
-                guest_conf = guest_conf/10 # convert to md_units
-                guest_conf = recenter(guest_conf, conf_center)
-
-                x0 = np.concatenate([host_conf, guest_conf])       # combined geometry
-
-                combined_pdb = Chem.CombineMols(Chem.MolFromPDBFile(host_pdb_file, removeHs=False), init_mol)
-                combined_pdb_str = StringIO(Chem.MolToPDBBlock(combined_pdb))
-                out_file = os.path.join("frames", "epoch_"+str(epoch)+"_"+mode+"_"+host_name+"_conf_"+str(conf_idx)+".pdb")
-                writer = PDBWriter(combined_pdb_str, out_file)
-                # set this to None if we don't care about visualization
-                all_args.append((x0, writer,  conf_idx % num_gpus, precision))
-
-            results = pool.map(sim.run_forward_multi, all_args)
-            all_du_dls.append(results)
+
+    lr = 5e-4
+    # opt_init, opt_update, get_params = optimizers.adam(lr)
+    opt_init, opt_update, get_params = optimizers.sgd(lr)
```

Comment:
I guess we will switch to adam later right?

Reply:
for batched, yes